scripts/dive_cpplint_wrapper.py

```diff
@@ -0,0 +1,154 @@
+"""cpplint wrapper for Dive."""
+
+import argparse
+import pathlib
+import subprocess
+
+DIVE_SOURCE_DIRS_CURRENT = (
+    "cli", "common", "dive_core", "gfxr_dump_resources", "gfxr_ext", "gpu_time",
+    "host_cli", "layer", "lrz_validator", "network", "plugins", "runtime_layer",
+    "tests", "trace_stats", "ui", "utils",
+)
+# Note: this is part of source tree organization
+DIVE_SOURCE_DIRS_FUTURE = ("include", "src")
+DIVE_SOURCE_DIRS = DIVE_SOURCE_DIRS_CURRENT + DIVE_SOURCE_DIRS_FUTURE
+DIVE_SOURCE_DIRS_SET = frozenset(DIVE_SOURCE_DIRS)
+DIVE_CPPLINT_FILE_SUFFIX = (".cpp", ".cc", ".h")
+DIVE_CPPLINT_FILTERS = (
+    # Dive repo specific
+    "-build/c++17",  # <filesystem>
+    "-readability/todo",
+    "-readability/braces",  # macro parsing with {} on next line
+    "-whitespace/braces",
+    "-whitespace/indent",
+    "-whitespace/newline",
+    "-whitespace/blank_line",
+    "-whitespace/parens",
+    # Enable later
+    "-build/include_what_you_use",
+    "-build/include_subdir",
+    "-build/include_order",
+    "-runtime/explicit",
+    "-readability/casting",
+    "-whitespace/comments",
+    # To be fixed soon
+    "-whitespace/line_length",
+    "-readability/namespace",
+    "-build/namespaces/source/namespace/nonliterals",
+    "-whitespace/ending_newline",
+    "-readability/inheritance",
+)
+
+DIVE_CPPLINT_FILTERS_EXT = tuple()
+DIVE_CPPLINT_FILTERS_G3 = (
+    "-build/c++11",
+    "-build/include",
+    "-build/header_guard",
+    "-runtime/deprecated_fn",
+)
+
+DIVE_CPPLINT_FLAGS = (
+    "--linelength=100",
+    "--filter="+",".join(DIVE_CPPLINT_FILTERS + DIVE_CPPLINT_FILTERS_EXT),
+)
+
+DIVE_CPPLINT_FLAGS_G3 = (
+    "--filter="+",".join(DIVE_CPPLINT_FILTERS + DIVE_CPPLINT_FILTERS_G3),
+)
+
+
+def list_files_pathlib_rglob(top_dirs, dive_root):
+    files = []
+    for top_dir in top_dirs:
+        for path in dive_root.joinpath(top_dir).rglob("*"):
+            if not path.is_file():
+                continue
+            if path.suffix not in DIVE_CPPLINT_FILE_SUFFIX:
+                continue
+            files.append(path)
+    return files
+
+
+def require_lint(path, dive_root_resolved):
+    """Check if current file need to run through cpplint."""
+    if not path.is_file():
+        return False
+    if not path.exists():
+        print("Skip non-exist file:", path)
+        return False
+    if not path.resolve().is_relative_to(dive_root_resolved):
+        print("WARNING: Found out of tree file:", path)
+        return False
+    rel_path = path.resolve().relative_to(dive_root_resolved)
+    if path.suffix not in DIVE_CPPLINT_FILE_SUFFIX:
+        print("Skip non-C++ file:", path)
+        return False
+    if not rel_path.parts or rel_path.parts[0] not in DIVE_SOURCE_DIRS_SET:
+        print("Skip non-Dive code:", path)
+        return False
+    return True, ""
```

Comment:
The second element of the returned tuple is unused and can be removed
```suggestion
    return True
```

Reply:
Done